traffic_ops/traffic_ops_golang/riaksvc/riak_services.go

```diff
@@ -247,34 +256,79 @@ func RiakServersToCluster(servers []ServerAddr, authOptions *riak.AuthOptions) (
 	if err != nil {
 		return nil, errors.New("creating riak cluster: " + err.Error())
 	}
+	return cluster, err
+}
+
+func RiakServersToCluster(servers []ServerAddr, authOptions *riak.AuthOptions) (StorageCluster, error) {
+	cluster, err := riakServersToClusterPriv(servers, authOptions)
+	if nil != err {
+		return nil, err
+	}
+	return RiakStorageCluster{Cluster: cluster}, nil
+}
+
+func GetPooledClusterTx(tx *sql.Tx, authOptions *riak.AuthOptions) (StorageCluster, error) {
+
+	var cluster *riak.Cluster
+	var err error
+
+	clusterMutex.Lock()
```

Comment:
I think you'd be safe to follow this up with a `defer clusterMutex.Unlock()` rather than explicitly unlocking at every return point.